binaries/runtime/src/main.rs

```diff
@@ -90,13 +83,15 @@ async fn run(
     mut operator_stop_publishers: HashMap<OperatorId, Box<dyn Publisher>>,
 ) -> eyre::Result<()> {
     #[cfg(feature = "metrics")]
-    let _started = init_meter();
-
-    #[cfg(feature = "metrics")]
-    let meter = global::meter(Box::leak(node_id.to_string().into_boxed_str()));
-
-    #[cfg(feature = "metrics")]
-    init_process_observer(meter);
+    {
+        use dora_metrics::init_meter;
+        use opentelemetry::global;
+        use opentelemetry_system_metrics::init_process_observer;
+
+        let _started = init_meter();
+        let meter = global::meter(Box::leak(node_id.to_string().into_boxed_str()));
```

Comment:
The problem with putting global::meter in a conditional feature block is that it is not going to live long enough. To record metrics during the lifetime of the `dora-runtime`. I think that it is going to get dropped at the end of the block, but let me know if it's not the case.

Reply:
Ah, that could be, good catch! So `meter` should live for the whole `run` method? What about `_started`?

Reply:
Hmm, seems like the `init_process_observer` function takes the `meter` by value, so it is not live after the call anymore. Did you perhaps mean the `_started` binding?